Prog/Demographic trends.sas

```diff
@@ -0,0 +1,1036 @@
+   dbms=csv
+   replace;
+run;
+
+/*household number*/
+
+data households (where= (ucounty in("11001","24017","24021","24031","24033","51013","51059","51107","51153","51510","51600","51610","51683","51685" )));
+set NCDB.Ncdb_master_update;
+keep ucounty Jurisdiction numhhs7 numhhs8 numhhs9 numhhs0 numhhs1;
+
+  if ucounty in ("11001") then Jurisdiction =1;
+  if ucounty  in ("24017") then Jurisdiction =2;
+  if ucounty  in ("24021") then Jurisdiction =3;
+  if ucounty  in ("24031") then Jurisdiction =4;
+  if ucounty  in ("24033") then Jurisdiction =5;
+  if ucounty  in ("51013") then Jurisdiction =6;
+  if ucounty  in ("51059", "51600", "51610") then Jurisdiction =7;
+  if ucounty  in ("51107") then Jurisdiction =8;
+  if ucounty  in ("51153", "51683", "51685") then Jurisdiction =9; 
+  if ucounty  in ("51510") then Jurisdiction =10; 
+
+run;
+
+proc sort data=households;
+by ucounty;
+run;
+
+data COGSarea_2017 (where=(upuma in ("1100101", "1100102", "1100103", "1100104", "1100105", "2401600", "2400301", "2400302","2401001", "2401002", "2401003", "2401004", "2401005", "2401006", "2401007", "2401101", "2401102", "2401103", "2401104", "2401105", "2401106", "2401107", "5101301", "5101302", "5159301", "5159302", "5159303", "5159304", "5159305", "5159306", "5159307", "5159308", "5159309", "5110701", "5110702" , "5110703", "5151244", "5151245", "5151246", "5151255"
+                    ) and pernum =1 and gq in (1,2)));
+set Ipums.Acs_2017_dc Ipums.Acs_2017_md Ipums.Acs_2017_va;
+keep upuma Jurisdiction hhwt totalunits pernum gq;
+	%assign_jurisdiction; 
+totalunits=1;
+	run;
+proc sort data=COGSarea_2017;
+by upuma;
+run;
+
+data COGSvacant_2017 (where=(upuma in ("1100101", "1100102", "1100103", "1100104", "1100105", "2401600", "2400301", "2400302","2401001", "2401002", "2401003", "2401004", "2401005", "2401006", "2401007", "2401101", "2401102", "2401103", "2401104", "2401105", "2401106", "2401107", "5101301", "5101302", "5159301", "5159302", "5159303", "5159304", "5159305", "5159306", "5159307", "5159308", "5159309", "5110701", "5110702" , "5110703", "5151244", "5151245", "5151246", "5151255" )));
+set Ipums.Acs_2017_vacant_dc Ipums.Acs_2017_vacant_md Ipums.Acs_2017_vacant_va;
+
+	%assign_jurisdiction; 
+if vacancy in (1,2,3) then vacunit = 1; else vacunit = 0;
+run;
+
+data COGvacant2017;
+set COGSvacant_2017(keep=upuma VACANCY Jurisdiction vacunit where=(vacancy in (1,2,3)));
+run;
+
+proc sort data=COGSvacant_2017;
+by upuma;
+run;
+
+data vacantunits;
+merge COGSvacant_2017 COGSarea_2017;
+by upuma;
+run;
+
+proc summary data= vacantunits;
+	class Jurisdiction;
+	var totalunits vacunit;
+	weight hhwt;
+	output out = vacancyrate  sum=;
+
+	format Jurisdiction Jurisdiction. ;
+	run;
+
+data vacancyrate;
+set vacancyrate;
+vacrate= vacunit/totalunits;
+run;
+proc sort data=COGSarea_2017 ;
+by Jurisdiction;
+run;
+
+data hh_housing;
+merge COGSarea_2017 vacancyrate;
+by Jurisdiction;
+run;
```

Comment:
@maggiesu0725 similar issue with this merge. You want to make sure you are doing one-to-one merges rather than one-to-many, otherwise it causes problems later on.